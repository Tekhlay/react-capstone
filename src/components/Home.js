/* eslint-disable max-len */
/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { FaArrowAltCircleRight } from 'react-icons/fa';
import { fetchCrypto } from '../redux/crypto/crypto';

const Home = () => {
  const [search, setSearch] = useState('');
  const coins = useSelector((state) => state.cryptoData);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchCrypto());
  }, [dispatch]);
  const handleSearch = (e) => {
    setSearch(e.target.value);
  };

  const filteredCoins = coins.crypto.filter((coin) => (
    coin.name.toLowerCase().includes(search.toLowerCase())
  ));

  return (
    <header className="home-container">
      <div className="home-serch">
        <input
          type="text"
          value={search}
          placeholder="Search Coins..."
          onChange={handleSearch}
        />
      </div>

      <div className="home-list">
        {coins.isLoding && <h1>Loading...</h1>}
        {coins.isEror && <h1>Something went wrong...</h1>}
        {!filteredCoins.length && <h1><em>No related coins found!</em></h1>}
        {filteredCoins.map((coin) => (
          <div key={coin.id} className="container-card">
            <h1>{coin.name}</h1>
            <img src={coin.icon} alt={coin.name} />
            <h1>{coin.price}</h1>
            <NavLink to={`/details/${coin.id}`}>
              <FaArrowAltCircleRight />
            </NavLink>
          </div>
        ))}
      </div>
    </header>
  );
};

export default Home;
