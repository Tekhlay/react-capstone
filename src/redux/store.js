/* eslint-disable import/no-extraneous-dependencies */
import { configureStore } from '@reduxjs/toolkit';
import cryptoReducer from './crypto/crypto';

const store = configureStore({
  reducer: {
    crypto: cryptoReducer,
  },
});

export default store;
