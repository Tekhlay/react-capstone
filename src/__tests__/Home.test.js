import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Home from '../components/Home';
import store from '../redux/store';

describe('Home', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Home />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
