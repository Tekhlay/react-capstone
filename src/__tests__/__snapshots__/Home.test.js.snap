// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<header
  className="home-container"
>
  <div
    className="home-serch"
  >
    <input
      onChange={[Function]}
      placeholder="Search Coins..."
      type="text"
      value=""
    />
  </div>
  <div
    className="home-list"
  >
    <h1>
      No related coins found!
    </h1>
  </div>
</header>
`;
